scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/SchedulingDSLCompilationService.java

```diff
@@ -0,0 +1,97 @@
+package gov.nasa.jpl.aerie.scheduler.server.services;
+
+import gov.nasa.jpl.aerie.json.JsonParser;
+import gov.nasa.jpl.aerie.scheduler.server.http.InvalidEntityException;
+import gov.nasa.jpl.aerie.scheduler.server.http.InvalidJsonException;
+import gov.nasa.jpl.aerie.scheduler.server.models.SchedulingDSL;
+import org.json.JSONObject;
+
+import javax.json.Json;
+import javax.json.stream.JsonParsingException;
+import java.io.File;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.List;
+
+public class SchedulingDSLCompilationService {
+
+  private final Process nodeProcess;
+
+  public SchedulingDSLCompilationService() throws SchedulingDSLCompilationException, IOException {
+    final var schedulingDslCompilerRoot = System.getenv("SCHEDULING_DSL_COMPILER_ROOT");
+    final var schedulingDslCompilerCommand = System.getenv("SCHEDULING_DSL_COMPILER_COMMAND");
+    this.nodeProcess = new ProcessBuilder("node", schedulingDslCompilerCommand)
+        .redirectError(ProcessBuilder.Redirect.INHERIT)
+        .directory(new File(schedulingDslCompilerRoot))
+        .start();
+
+    final var inputStream = this.nodeProcess.outputWriter();
+    inputStream.write("ping\n");
+    inputStream.flush();
+    if (!"pong".equals(this.nodeProcess.inputReader().readLine())) {
```

Comment:
I know this is pretty minor, but it feel uncomfortable to have the hardcoded string first here. I guess it just feels more natural to say "hey, this thing here should be equal to this value", rather than "this value should be equal to this thing here."

Reply:
I agree - the main reason to put the string literal first is to avoid a null pointer exception if the value is `null`.

Reply:
There's a method provided by `Objects` that handles the null case - I'm okay with writing it this way if you find it more readable:

```suggestion
    if (!Objects.equals(this.nodeProcess.inputReader().readLine(), "pong")) {
```

Reply:
That's a good point, I hadn't thought of that. I do like the `Objects.equals` way.

Reply:
Done. Latest force push uses the proposed `Objects.equals` instead of string literal `.equals()`.